RedditSharp/Exceptions/RedditException.cs

```
@@ -8,6 +9,8 @@ namespace RedditSharp
     [Serializable]
     public class RedditException : Exception
     {
+        public JArray Errors { get; }
+
```

Does this actually return the errors from Reddit properly? That's been something I've been meaning to fix up. It also needs an HTTP status code preferably.

---

Yes, it does, but only for the new endpoints I've implemented. But yeah, I agree the code in `ExecuteRequestAsync` needs to be cleaned up to expose the errors instead of swallowing an exception.

Example response when trying to delete a collection that doesn't exist: `{"json": {"errors": [["INVALID_COLLECTION_ID", "That collection doesn't exist", "collection_id"]]}}` but the status code is `200`. 

"errors" is an array of arrays... kinda weird, but the consumer can do whatever with it

---

@CrustyJew It appears RedditHttpException has the status code built in. Perhaps eventually having an `Error` class that can be used across the board?

---

@chasedog @jkrejcha I think it would be better to get these rolled together so you have one exception type and can determine if it is an HTTP status code issue or a reddit result issue.

---

The great thing about the reddit API is that it seems to be a combination of both, either one, none, or sometimes a completely different error code dependent on whether you're accessing via the site or via the JSON API. There does need to be handling of errors of some sort. RedditHttpException should probably inherit from RedditException IMO but it's hard to say.